src/client/datascience/ipywidgets/cdnWidgetScriptSourceProvider.ts

```diff
@@ -81,53 +81,59 @@ export class CDNWidgetScriptSourceProvider implements IWidgetScriptSourceProvide
         this.cache.clear();
     }
     public async getWidgetScriptSource(moduleName: string, moduleVersion: string): Promise<WidgetScriptSource> {
+        // First see if we already have it downloaded.
```

Comment:
I believe this solves the issue with ipywidgets (or at least one of them).
The problem was we were download the widget file twice.
First time we download its all good.
Then IPyWidgets starts to fetch the file from the webview.
While webview is fetching the file from disc (that was already downloaded), 
- the extension is downloading the file and overwriting the local disc file again

Thus, we end up overwriting a file that is being read by another process.
& the ipywidget webview then fails to download the file & kaboom...